src/ResourceManager/Kubernetes/Commands.Kubernetes/Commands/CreateOrUpdateKubeBase.cs

```diff
@@ -0,0 +1,361 @@
+            HelpMessage = "Resource Group Name.")]
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = SpParamSet,
+            HelpMessage = "Resource Group Name.")]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = DefaultParamSet,
+            HelpMessage = "Kubernetes managed cluster Name.")]
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = SpParamSet,
+            HelpMessage = "Kubernetes managed cluster Name.")]
+        [ValidateNotNullOrEmpty]
+        [ValidatePattern("^[a-zA-Z0-9][a-zA-Z0-9_.-]*$")]
+        [ValidateLength(2, 64)]
+        public string Name { get; set; }
+
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ParameterSetName = SpParamSet,
+            HelpMessage = "The client id and client secret associated with the AAD application / service principal.")]
+        public PSCredential ClientIdAndSecret { get; set; }
+
+        [Parameter(Mandatory = false,
+            HelpMessage = "Azure location for the cluster. Defaults to the location of the resource group.")]
+        [LocationCompleter("Microsoft.ContainerService/managedClusters")]
+        public string Location { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "User name for the Linux Virtual Machines.")]
+        public string AdminUserName { get; set; } = "azureuser";
+
+        [Parameter(Mandatory = false, HelpMessage = "The DNS name prefix for the cluster.")]
+        public string DnsNamePrefix { get; set; }
+
+
+        [Parameter(Mandatory = false, HelpMessage = "The version of Kubernetes to use for creating the cluster.")]
+        [PSArgumentCompleter("1.7.7", "1.8.1")]
+        public string KubernetesVersion { get; set; } = "1.8.1";
+
+        [Parameter(Mandatory = false, HelpMessage = "The default number of nodes for the node pools.")]
+        public int NodeCount { get; set; } = 3;
+
+        [Parameter(Mandatory = false, HelpMessage = "The default number of nodes for the node pools.")]
+        public int NodeOsDiskSize { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "The size of the Virtual Machine.")]
+        public string NodeVmSize { get; set; } = "Standard_D2_v2";
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "SSH key file value or key file path. Defaults to {HOME}/.ssh/id_rsa.pub.")]
+        [Alias("SshKeyPath")]
+        public string SshKeyValue { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public Hashtable Tags { get; set; }
```

Comment:
Rename to "Tag" (more info [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#singular-noun-1))